framework/i18n/Formatter.php

```diff
@@ -1174,18 +1244,157 @@ public function asSize($value, $decimals = null, $options = [], $textOptions = [
+     * @param array $textOptions optional configuration for the number formatter. This parameter will be merged with [[numberFormatterTextOptions]].
+     * @return string the formatted result.
+     * @throws InvalidParamException if the input value is not numeric or the formatting failed.
+     * @see asLength
+     */
+    public function asShortLength($value, $baseUnit = null, $system = null, $decimals = null, $options = [], $textOptions = [])
+    {
+        return $this->formatUnit(self::UNIT_LENGTH, self::FORMAT_WIDTH_SHORT, $value, $baseUnit, $system, $decimals, $options, $textOptions);
+    }
+
+    /**
+     * Formats the value as a weight in human readable form for example `12 kilograms`.
+     *
+     * @param integer $value value to be formatted.
+     * @param double $baseUnit unit of value as the multiplier of the smallest unit
+     * @param string $system either self::UNIT_SYSTEM_METRIC or self::UNIT_SYSTEM_IMPERIAL, if null defaults to metric
+     * @param integer $decimals the number of digits after the decimal point.
+     * @param array $options optional configuration for the number formatter. This parameter will be merged with [[numberFormatterOptions]].
+     * @param array $textOptions optional configuration for the number formatter. This parameter will be merged with [[numberFormatterTextOptions]].
+     * @return string the formatted result.
+     * @throws InvalidParamException if the input value is not numeric or the formatting failed.
+     * @see asWeight
+     */
+    public function asWeight($value, $baseUnit = null, $system = null, $decimals = null, $options = [], $textOptions = [])
+    {
+        return $this->formatUnit(self::UNIT_WEIGHT, self::FORMAT_WIDTH_LONG, $value, $baseUnit, $system, $decimals, $options, $textOptions);
+    }
+
+    /**
+     * Formats the value as a weight in human readable form for example `12 kg`.
+     *
+     * This is the short form of [[asWeight]].
+     *
+     * @param double $value value to be formatted.
+     * @param double $baseUnit unit of value as the multiplier of the smallest unit
+     * @param string $system either self::UNIT_SYSTEM_METRIC or self::UNIT_SYSTEM_IMPERIAL, if null defaults to metric
+     * @param integer $decimals the number of digits after the decimal point.
+     * @param array $options optional configuration for the number formatter. This parameter will be merged with [[numberFormatterOptions]].
+     * @param array $textOptions optional configuration for the number formatter. This parameter will be merged with [[numberFormatterTextOptions]].
+     * @return string the formatted result.
+     * @throws InvalidParamException if the input value is not numeric or the formatting failed.
+     * @see asWeight
+     */
+    public function asShortWeight($value, $baseUnit = null, $system = null, $decimals = null, $options = [], $textOptions = [])
```

Comment:
these methods have a bit too many options, which would override global formatter settings. Should only allow overriding those that are commonly expected to differ from global settings.